run_task.py

```diff
@@ -3,46 +3,84 @@
 import time
 from typing import List
 
+import click
 import pandas as pd
 import vantage6.client as vtgclient
 
-USERNAME = 'admin'
-PASSWORD = 'admin'
-
-POST = 'POST'
-WAIT_TIME = 1
-RETRIES = 20
-
-HOST = 'http://localhost'
-PORT = 5001
-
-IMAGE = 'localhost:5000/v6-carrier-py'
-METHOD = 'correlation_matrix'
-COLLABORATION_ID = 1
-ORGANIZATION_IDS = [1]  # [2, 3, 6]
-MASTER = True
-NUM_NODES = 1
-
-
-def main():
-    client = vtgclient.Client(HOST, PORT)
-    client.authenticate(USERNAME, PASSWORD)
+DEFAULT_USERNAME = 'admin'
+DEFAULT_PASSWORD = 'admin'
+
+DEFAULT_WAIT_TIME = 1
+DEFAULT_NUM_RETRIES = 20
+
+DEFAULT_HOST = 'http://localhost'
+DEFAULT_PORT = 5001
+
+DEFAULT_IMAGE = 'localhost:5000/v6-carrier-py'
+DEFAULT_METHOD = 'correlation_matrix'
+DEFAULT_COLLABORATION_ID = '1'
+DEFAULT_ORGANIZATION_IDS = '1'
+
+
+@click.command(context_settings={'ignore_unknown_options': True,
+                                 'allow_extra_args': True},
+               help='Run a task on vantage6 nodes. Optionally pass extra'
+                    'keyword arguments to be passed to the algorithm'
+                    '(i.e. "--key value")')
+@click.option('--method', default=DEFAULT_METHOD,
+              help='Method to run')
+@click.option('--image', default=DEFAULT_IMAGE,
+              help='Docker image to run')
+@click.option('--collaboration_id', default=DEFAULT_COLLABORATION_ID,
+              help='Identifier for collaboration to run task with', type=int)
+@click.option('--organization_ids', default=DEFAULT_ORGANIZATION_IDS,
+              help='Identifier for organization ids that should run task, '
+                   'pass as comma-separated string ("1,2")')
+@click.option('--username', default=DEFAULT_USERNAME)
+@click.option('--password', default=DEFAULT_PASSWORD)
+@click.option('--host', default=DEFAULT_HOST,
+              help='Host for vantage6 server')
+@click.option('--port', default=DEFAULT_PORT, type=int,
+              help='Port of vantage6 server')
+@click.option('--wait_time', default=DEFAULT_WAIT_TIME, type=int,
+              help='Time in seconds to wait in between polling tries')
+@click.option('--num_retries', default=DEFAULT_NUM_RETRIES, type=int,
+              help='Number of retries for polling task results')
+@click.option('--master/--rpc', default=True,
```

Comment:
How does this option work? Does this create a boolean value based on whether you specify `--master` or `--rpc`?

Reply:
I've added a comment. What you assumed is correct. It's a bit of an implicit `click` thing maybe, we could also use a flag? Or change `--rpc` into `--no-master`.